Zend/zend.h

```diff
@@ -24,21 +24,18 @@
 
 #define ZEND_ENGINE_3
 
-#include "zend_types.h"
-#include "zend_map_ptr.h"
-#include "zend_errors.h"
-#include "zend_alloc.h"
-#include "zend_llist.h"
-#include "zend_string.h"
-#include "zend_hash.h"
-#include "zend_ast.h"
-#include "zend_gc.h"
-#include "zend_variables.h"
-#include "zend_iterators.h"
-#include "zend_stream.h"
-#include "zend_smart_str_public.h"
-#include "zend_smart_string_public.h"
-#include "zend_signal.h"
+#include "zend_error.h" // TODO remove
+#include "zend_map_ptr.h" // for ZEND_MAP_PTR_DEF()
+#include "zend_errors.h" // for E_* (as zend_error_noreturn() parameter)
+#include "zend_portability.h" // for BEGIN_EXTERN_C
+#include "zend_signal.h" // for ZEND_SIGNAL_[UN]BLOCK_INTERRUPTIONS()
+#include "zend_smart_str_public.h" // for smart_str
```

Comment:
Not sure this sort of comment adds value here.

Reply:
These comments have been critticized a lot, yet I only imitated what I already found in the PHP code base, e.g.:
https://github.com/php/php-src/blob/3ed526441400060aa4e618b91b3352371fcd02a8/main/streams/streams.c#L29-L30
https://github.com/php/php-src/blob/3ed526441400060aa4e618b91b3352371fcd02a8/sapi/cli/php_cli_server.c#L93-L98
https://github.com/php/php-src/blob/3ed526441400060aa4e618b91b3352371fcd02a8/ext/fileinfo/fileinfo.c#L30-L33
https://github.com/php/php-src/blob/3ed526441400060aa4e618b91b3352371fcd02a8/ext/hash/xxhash/xxhash.h#L342
https://github.com/php/php-src/blob/3ed526441400060aa4e618b91b3352371fcd02a8/ext/hash/xxhash/xxhash.h#L1485
https://github.com/php/php-src/blob/3ed526441400060aa4e618b91b3352371fcd02a8/ext/hash/xxhash/xxhash.h#L5448
https://github.com/php/php-src/blob/3ed526441400060aa4e618b91b3352371fcd02a8/ext/mysqlnd/config-win.h#L11
https://github.com/php/php-src/blob/3ed526441400060aa4e618b91b3352371fcd02a8/ext/standard/info.c#L30
https://github.com/php/php-src/blob/3ed526441400060aa4e618b91b3352371fcd02a8/ext/standard/php_http.h#L21
https://github.com/php/php-src/blob/3ed526441400060aa4e618b91b3352371fcd02a8/ext/standard/formatted_print.c#L17
I think those comments are helpful, but [my RFC](https://wiki.php.net/rfc/include_cleanup) will hopefully lead to a conclusion whether these comments are desired. If they are desired, I'll keep them in my PR; if not, I'll remove them, and submit a PR to remove all existing `#include` comments.

(Another thing to decide is whether forward declarations are allowed; the one guy who criticized me most for adding them happens to be somebody who added many existing forward declarations to the PHP code base; such double standards confuse me, that's annoying.)

Reply:
My personal position about all of this is more "whatever". If your proposal passes then let 's merge this, if it doesn't then it doesn't. It is not particularly a fundamental nor greatly important kind of changes in my view, unless ... it brings value like decreasing the build time. Granted, compiling php is nowhere near compiling Qt 🙂 but any help is valuable. I m more indifferent to the other "benefits".

Reply:
@devnexen It may not seem like this changes would have a big impact, but for someone who wants to learn how PHP works under the hood, so they may eventually contribute, it adds a lot of value to clean up the code base.